src/Compat/SearchControls.php

```diff
@@ -58,10 +58,13 @@ public function createSearchBar(Query $query, ...$params): SearchBar
         $preserveParams = array_pop($params) ?? [];
         $redirectUrl = array_pop($params);
 
+        $paramsToAdd = $requestUrl->onlyWith($preserveParams)->getParams()->toArray(false);
```

Comment:
Only `addParams` requires an actual array. Don't transform parameters needlessly to an array.